pkg/roachprod/testselector/sfselector/selector.go

```diff
@@ -0,0 +1,307 @@
+// Copyright 2024 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+// test selector is responsible for selecting the tests for running in the roachtest
+// it generates a csv file with test details
+
+package sfselector
+
+import (
+	"bytes"
+	"context"
+	gosql "database/sql"
+	"fmt"
+	"io"
+	"strings"
+
+	secretmanager "cloud.google.com/go/secretmanager/apiv1"
+	"cloud.google.com/go/secretmanager/apiv1/secretmanagerpb"
+	"cloud.google.com/go/storage"
+	"github.com/cockroachdb/cockroach/pkg/util/encoding/csv"
+	sf "github.com/snowflakedb/gosnowflake"
+	"google.golang.org/api/option"
+)
+
+const (
+	account   = "lt53838.us-central1.gcp"
+	database  = "DATAMART_PROD"
+	schema    = "TEAMCITY"
+	warehouse = "COMPUTE_WH"
+
+	// bucket and testsFileLocation are for uploading the csv for each cloud in GCS
+	bucket            = "roachtest-data"
```

Comment:
Same comment as before, please don't use a bucket in `cockroach-ephemeral`, use one in a project owned wholly by test eng.

Reply:
I moved the bucket!